src/server/game/Server/WorldSession.cpp

```diff
@@ -926,7 +927,7 @@ void WorldSession::ReadMovementInfo(WorldPacket &data, MovementInfo* mi)
             mi->RemoveMovementFlag((maskToRemove));
     #endif
 
-    Unit* mover = GetGameClient()->GetActivelyMovedUnit();
+    Unit* mover = GetPlayer()->GetGUID() == mi->guid ? GetPlayer() : ObjectAccessor::GetUnit(*GetPlayer(), mi->guid);
```

Comment:
could you please add an if check on mover after this line and handle that case too ? whether it's return + log error or ASSERT (even if a return + log is always better if possible so it doesn't crash the whole server)

Reply:
mi->guid is client controlled, it must not be ASSERT

Reply:
@jackpoz will do tonight.

@Shauren 

> how does this change affect cheating prevention? 

WorldSession::ReadMovementInfo is called in a movement handler only after all the permission checks are made

Reply:
Wouldnt this mean that there is a data consistency error in GameClient state? `_allowedMovers` has a guid but `_activelyMovedUnit` is not set?

Reply:
@jackpoz done. i choose the assert approach and not just returning because otherwise weird things will happen after the function will return.

@Shauren It might. I'll look into it. It could also be explained by the following edge case:
on login, there is a window of time (after the server initialize the player and sends `CreateObject2` to the client and before the clients sends `CMSG_SET_ACTIVE_MOVER`) during which the GameClient's `_allowedMovers` contains the player but `_activelyMovedUnit` is null. If the client sends an ACK message during that time window, we get the situation you described

Reply:
In the meantime, let's merge this already?

Reply:
Did you check the crashlogs mentioned below ?